code/modules/antagonists/changeling/powers/mutations.dm

```diff
@@ -463,6 +463,61 @@
 	if(ismob(loc))
 		loc.visible_message("<span class='warning'>[loc.name]\'s flesh turns black, quickly transforming into a hard, chitinous mass!</span>", "<span class='warning'>We harden our flesh, creating a suit of armor!</span>", "<span class='warning'>You hear organic matter ripping and tearing!</span>")
 
+
+/obj/item/clothing/suit/armor/changeling/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text, final_block_chance, damage, attack_type)
+	. = ..()
+
+	if(!IS_HORIZONTAL(owner))
+		return
+
+	var/obj/item/I = hitby
+	if(!istype(I))
+		return
+	if(!istype(I.loc, /mob/living))
+		return
+
+	var/mob/living/user = I.loc
+
+	// snowflake checks my beloved
+	// this will become tooltype checks I swear
+	if(!istype(I, /obj/item/circular_saw) && !istype(I, /obj/item/twohanded/required/chainsaw) && !istype(I, /obj/item/twohanded/chainsaw))
+		return
+
+	user.visible_message(
+		"<span class='notice'>[user] starts to saw through [owner]'s [src].</span>",
+		"<span class='notice'>You start to saw through [owner]'s [src].</span>",
+		"<span class='notice'>You hear a loud grinding noise.</span>"
+	)
+
+	if(!do_after(user, 15 SECONDS, target = owner))
+		user.visible_message(
+			"<span class='warning'>[user] fails to cut through [owner]'s [src].</span>",
+			"<span class='warning'>You fail to cut through [owner]'s [src].</span>",
+			"<span class='notice'>You hear the grinding end.</span>"
+		)
+		return FALSE
+
+	if(!IS_HORIZONTAL(owner))
+		return FALSE
+	user.visible_message(
+		"<span class='warning'>\The [src] turns to shreds as [user] cleaves through it.</span>",
+		"<span class='warning'>\The [src] turns to shreds as you cleave through it.</span>",
+		"<span class='notice'>You hear something fall as the grinding ends.</span>"
+	)
+
+	playsound(I, I.hitsound, 50)
+	// you've torn it up, get rid of it.
+	new /obj/effect/decal/cleanable/shreds(owner.loc)
+	owner.unEquip(src, TRUE, TRUE)
+	if(istype(owner.head, /obj/item/clothing/head/helmet/changeling))
+		var/head = owner.head
```

Comment:
you can probably just drop this var declaration and just use `owner.head` since you're not going to be using it for more than 1-2 lines (probably only 1)